packages/shared/src/components/onboarding/FilterOnboardingV4.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { ReactElement, useMemo } from 'react';
+import classNames from 'classnames';
+import { useMutation, useQuery, useQueryClient } from 'react-query';
+import request from 'graphql-request';
+import useFeedSettings from '../../hooks/useFeedSettings';
+import { RequestKey, generateQueryKey } from '../../lib/query';
+import {
+  GET_ONBOARDING_TAGS_QUERY,
+  GET_RECOMMENDED_TAGS_QUERY,
+  Tag,
+  TagsData,
+} from '../../graphql/feedSettings';
+import { graphqlUrl } from '../../lib/config';
+import { disabledRefetch } from '../../lib/func';
+import { Button } from '../buttons/Button';
+import { AlertColor, AlertDot } from '../AlertDot';
+import { SearchField } from '../fields/SearchField';
+import useDebounce from '../../hooks/useDebounce';
+import { useTagSearch } from '../../hooks';
+import type { FilterOnboardingProps } from './FilterOnboarding';
+import useTagAndSource from '../../hooks/useTagAndSource';
+import { Origin } from '../../lib/analytics';
+
+type OnSelectTagProps = {
+  tag: Tag;
+};
+
+const tagsSelector = (data: TagsData) => data?.tags || [];
+
+export function FilterOnboardingV4({
+  className,
+}: FilterOnboardingProps): ReactElement {
+  const queryClient = useQueryClient();
+
+  const { feedSettings } = useFeedSettings();
+  const selectedTags = useMemo(() => {
+    return new Set(feedSettings?.includeTags || []);
```

Comment:
We also have `uniqueifyArray` or use `Array.from(new Set(includeTags))`
No strong preference, but easier to identify if we ever want to consider uniforming stuff, or changing every type to something.

Reply:
I want the Set here specifically for faster checks vs array.

Reply:
Ah I see :) Ok then let's leave it.
Set.has is faster then array methods?

Reply:
Yes, Set has O(1) access.